packages/service/model/request.ts

```diff
@@ -0,0 +1,64 @@
+import { Type } from '@sinclair/typebox';
+import { RequestId, UserId, CourseId, FileReference } from './util.js';
+
+export const RequestDetails = Type.Object({
+    reason: Type.String(),
+    proof: Type.Array(FileReference),
+});
+
+export const Response = Type.Object({
+    instructorId: UserId,
+    timestamp: Type.Date(),
+    approved: Type.Boolean(),
+    remark: Type.String(),
+});
+
+export const BaseRequest = Type.Object({
+    id: RequestId,
+    type: Type.String(),
+    studentId: UserId,
+    courseId: CourseId,
+    timestamp: Type.Date(),
+    metadata: Type.Object({}, { additionalProperties: true }),
+    details: RequestDetails,
+    response: Type.Union([Response, Type.Null()]),
+});
+
+/* Below are actual requests */
+/* Description of fields under metadata can be used at frontend */
+
+export const SwapSectionRequest = Type.Intersect([
```

Comment:
```suggestion
export const SwapSectionRequest = Type.Composite([
```

如果要 merge 两个不重叠的 object 的话用 composite 比较好，就相当于你把那些 object properties 写在一起。

Reply:
刚看到你上面回的了，那就还是用 intersect 比较好